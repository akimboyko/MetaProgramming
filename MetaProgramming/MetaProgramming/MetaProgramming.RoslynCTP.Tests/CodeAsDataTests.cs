using System;
using System.Collections.Generic;
using System.IO;
using ApprovalTests;
using ApprovalTests.Reporters;
using Newtonsoft.Json;
using NUnit.Framework;

namespace MetaProgramming.RoslynCTP.Tests
{
    [TestFixture]
    [UseReporter(typeof(DiffReporter))]
    public class CodeAsDataTests
    {
        [Test]
        public void ScriptAsData_ConvertJsonToScript_ScriptSubmission()
        {
            // Arrange
            var scriptInfo = JsonConvert.DeserializeObject<ScriptInfo>(File.ReadAllText(@"./ScriptInfo.json"));
            var dataClassesInfo = JsonConvert.DeserializeObject<ClassTemplateInfo[]>(File.ReadAllText(@"./ClassTemplateInfos.json"));
            Func<Type, object> deserializeToType = type => JsonConvert.DeserializeObject(File.ReadAllText(@"./InputData.json"), type.MakeArrayType());

            // Act
            IEnumerable<dynamic> results = CodeAsData.ProcessScript(scriptInfo, dataClassesInfo, deserializeToType);

            // Assert
            Approvals.Verify(JsonConvert.SerializeObject(results, Formatting.Indented));
        }
    }
}
