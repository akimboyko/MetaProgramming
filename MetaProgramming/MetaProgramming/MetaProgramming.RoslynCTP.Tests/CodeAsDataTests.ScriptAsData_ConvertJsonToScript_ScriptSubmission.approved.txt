[
  {
    "Σ": 10.2500,
    "Δ": 0.2500,
    "λ": "Some description"
  },
  {
    "Σ": 20.0400,
    "Δ": 0.0400,
    "λ": "Some description"
  },
  {
    "Σ": 12.2250,
    "Δ": 0.2250,
    "λ": "Some description"
  },
  {
    "Σ": 7.2000,
    "Δ": 7.2000,
    "λ": "Some description"
  }
]